request.js

```diff
@@ -0,0 +1,47 @@
+//var rp = require('request-promise');
+import requestPromise from 'request-promise';
+
+class Request {
+    constructor(method, url) {
+        this.method = method;
+        this.url = url;
+
+    }
+
+    static async performRequest(endpoint, method, req, data, api) {
```

Comment:
Due to context, you don't need to precise what you perform (execute is more appropriate)